manifests/config/passenger/fragment.pp

```diff
@@ -7,23 +7,25 @@
 #  $ssl_content:: content of the ssl virtual host fragment
 #
 define foreman::config::passenger::fragment(
-  $content='',
-  $ssl_content='',
+  $content=undef,
+  $ssl_content=undef,
 ) {
   require foreman::config::passenger
 
-  file { "${apache::confd_dir}/05-foreman.d/${name}.conf":
-    ensure  => present,
-    content => $content,
-    owner   => 'root',
-    group   => 'root',
-    mode    => '0644',
-    notify  => Class['apache::service'],
+  if $content {
```

Comment:
Ah, excellent point.  Yeah, it recursively purges: https://github.com/puppetlabs/puppetlabs-apache/blob/master/manifests/init.pp#L162-L163

I was wondering whether to add an `else` block that ensures the file is absent, but that's probably OK without it.